charms/worker/k8s/charmcraft.yaml

```
@@ -214,6 +214,25 @@ config:
       default: true
       description: |
         Enables or disables the network feature.
+    ingress-enabled:
+      type: boolean
+      default: false
+      description: |
+        Determines if the ingress feature should be enabled.
+    ingress-default-tls-secret:
+      type: string
+      default: ""
+      description: |
+        Sets the name of the secret to be used for providing default encryption to
+        ingresses.
```

Is this a kubernetes secret?  Can we clarify it's not to be confused with a `juju secret`?  Also, is there a namespace this secret should stay in?  Is it always kube-system or something else?

```suggestion
        Sets the name of the kubernetes secret to be used for providing default encryption to
        ingresses.
```

---

Thanks for pointing this out. I guess the namespace depends on the ingress being used.
For contour (in moonray impl) it's `projectcontour-root`: https://github.com/canonical/k8s-snap/blob/main/src/k8s/pkg/k8sd/features/contour/chart.go#L24-L29
For cilium (in default impl) it's `kube-system`: https://github.com/canonical/k8s-snap/blob/main/src/k8s/pkg/k8sd/features/cilium/ingress.go#L30-L38

Since I believe we're going with the default implementation which uses Cilium, it's best to say it is in the `kube-system`? Or maybe we can point them to a reference page about this.

---

yucky.  that's sad

if this secret moves between implementations we shouldn't specify which namespace.  We should be clear that's it is closely associated with the ingress implementation.  

Who sets this secret by the way?  Is the snap creating the secret (via the helm charm)?  Why do I want to set this secret name? I'm sorry i have so many questions. 

let's  make sure the description here is generic enough to handle multiple implementations under the hood, but specific enough to let the juju admin know they need to administer something in the cluster. 

if it's useless -- can we just drop the secret?

@bschimke95 do you have any feedback here?